src/core/Akka.FSharp/Spawning.fs

```diff
@@ -0,0 +1,140 @@
+    /// <param name="actorFactory">Either actor system or parent actor</param>
+    /// <param name="name">Name of spawned child actor</param>
+    /// <param name="f">Used by actor for handling response for incoming request</param>
+    /// <param name="options">List of options used to configure actor creation</param>
+    let spawnOpt (actorFactory : IActorRefFactory) (name : string) (f : Actor<'Message> -> Cont<'Message, 'Returned>) 
+        (options : SpawnOption list) : ActorRef<'Message> = 
+        let e = Linq.Expression.ToExpression(fun () -> new FunActor<'Message, 'Returned>(f))
+        let props = applySpawnOptions (Props.Create e) options
+        typed (actorFactory.ActorOf(props, name)) :> ActorRef<'Message>
+    
+    /// <summary>
+    /// Spawns an actor using specified actor computation expression.
+    /// The actor can only be used locally. 
+    /// </summary>
+    /// <param name="actorFactory">Either actor system or parent actor</param>
+    /// <param name="name">Name of spawned child actor</param>
+    /// <param name="f">Used by actor for handling response for incoming request</param>
+    let spawn (actorFactory : IActorRefFactory) (name : string) (f : Actor<'Message> -> Cont<'Message, 'Returned>) : ActorRef<'Message> = 
+        spawnOpt actorFactory name f []
+    
+    /// <summary>
+    /// Spawns an actor using specified actor quotation, with custom spawn option settings.
+    /// The actor can only be used locally. 
+    /// </summary>
+    /// <param name="actorFactory">Either actor system or parent actor</param>
+    /// <param name="name">Name of spawned child actor</param>
+    /// <param name="f">Used to create a new instance of the actor</param>
+    /// <param name="options">List of options used to configure actor creation</param>
+    let spawnObjOpt (actorFactory : IActorRefFactory) (name : string) (f : Quotations.Expr<unit -> #ActorBase>) 
+        (options : SpawnOption list) : ActorRef<'Message> = 
+        let e = Linq.Expression.ToExpression<'Actor> f
+        let props = applySpawnOptions (Props.Create e) options
+        typed (actorFactory.ActorOf(props, name)) :> ActorRef<'Message>
+    
+    /// <summary>
+    /// Spawns an actor using specified actor quotation.
+    /// The actor can only be used locally. 
+    /// </summary>
+    /// <param name="actorFactory">Either actor system or parent actor</param>
+    /// <param name="name">Name of spawned child actor</param>
+    /// <param name="f">Used to create a new instance of the actor</param>
+    let spawnObj (actorFactory : IActorRefFactory) (name : string) (f : Quotations.Expr<unit -> #ActorBase>) : ActorRef<'Message> = 
+        spawnObjOpt actorFactory name f []
+    
+    /// <summary>
+    /// Wraps provided function with actor behavior. 
+    /// It will be invoked each time, an actor will receive a message. 
+    /// </summary>
+    let actorOf (fn : 'Message -> unit) (mailbox : Actor<'Message>) : Cont<'Message, 'Returned> = 
```

Comment:
Would it be worth changing the signature of this just a bit? It's effectively re-arranging deck chairs for this one, but the direct consumer of `actorOf` is not usually going to provide a `mailbox`. What about:

    let actorOf (fn : 'Message -> unit) : (Actor<'Message> -> Cont<'Message, 'Returned>) =
        fun mbx ->
            let rec loop() = actor {
                let! msg = mbx.Receive()
                fn msg
                return! loop()
            }
        loop()

Similar with `actorOf2` below. Again, due to associativity, this doesn't change much, but it makes it more clear that what `actorOf` is creating for you is a function which takes a mailbox and returns a continuation, which is what the above `spawn` variations actually want.